packages/nuxt/src/pages/module.ts

```diff
@@ -237,10 +240,70 @@ export default defineNuxtModule({
     nuxt.hook('imports:extend', (imports) => {
       imports.push(
         { name: 'definePageMeta', as: 'definePageMeta', from: resolve(runtimeDir, 'composables') },
+        { name: 'defineRouteRules', as: 'defineRouteRules', from: resolve(runtimeDir, 'composables') },
         { name: 'useLink', as: 'useLink', from: '#vue-router' }
       )
     })
 
+    // Track mappings of absolute files to globs
+    let pageToGlobMap = {} as { [absolutePath: string]: string | null }
+    nuxt.hook('pages:extend', (pages) => { pageToGlobMap = getMappedPages(pages) })
+
+    // Extracted route rules defined inline in pages
+    const inlineRules = {} as { [glob: string]: NitroRouteConfig }
+    // User-provided route rules from `nuxt.config`
+    let baseRules: { [glob: string]: NitroRouteConfig } | undefined
+    nuxt.hook('nitro:config', (config) => { baseRules = config.routeRules })
+
+    // Allow telling Nitro to reload route rules
+    let updateRouteConfig: () => void | Promise<void>
+    nuxt.hook('nitro:init', (nitro) => {
+      updateRouteConfig = () => nitro.updateConfig({ routeRules: defu(inlineRules, baseRules) })
```

Comment:
One concern would be if other modules are also updating the route rules at runtime, it _seems_ like they would get overridden?

I think the scope for this happening, especially just in development, is pretty low.

Maybe worth merging in `nitro.options.routeRules` though?

Reply:
I think we can use `nitro.options._config.routeRules`.

cc: @pi0 - thoughts on how to avoid clobbering pre-existing route rules?

Reply:
I guess merging with `nuxt.options.routeRules` would be safer as it contains the latest user (nuxt module) modified rules. In context of Nuxt with top level `routeRules`, it should be main source of trust.

Reply:
nitro:config hook is last place for modules to modify nitro config, so that might be safer. (Hence current and previous implementations.)

Otherwise when we update the rules we overwrite intervening changes.

Or maybe use proxy for runtimeConfig/routeRules in resolved options that automatically reloads both within nitro?

Reply:
Modules might need to also apply route rules afterward as well (similar to runtimeConfig). We can either sync references so that nitro and nuxt modules update same object or proxy is also good idea for auto reload 👍🏼 (a possible downside is that sometimes we need to batch changes, advising to prefer update config hook is safer)

Reply:
How about instead letting nitro keep track of current state and only merge in/reload the _updated_ rules? That way we wouldn't overwrite anything.